internal/explorer/helpers.go

```diff
@@ -53,406 +50,170 @@ func errorReply(err error, w http.ResponseWriter) {
-	return twinID, nil
-}
-
-func (a *App) baseQuery(queryString string) (io.ReadCloser, error) {
-	jsonData := map[string]string{
-		"query": queryString,
-	}
-	jsonValue, err := json.Marshal(jsonData)
-	if err != nil {
-		return nil, fmt.Errorf("invalid query string %w", err)
-	}
 
-	request, err := http.NewRequest("POST", a.explorer, bytes.NewBuffer(jsonValue))
-	if err != nil {
-		return nil, fmt.Errorf("failed to query explorer network %w", err)
-	}
-
-	request.Header.Set("Content-Type", "application/json")
-	client := &http.Client{Timeout: time.Second * 10}
-	response, err := client.Do(request)
-	if err != nil {
-		return nil, fmt.Errorf("failed to query explorer network %w", err)
-	}
-
-	if response.StatusCode == 200 {
-		return response.Body, nil
-	}
-
-	var errResult interface{}
-	if err := json.NewDecoder(response.Body).Decode(errResult); err != nil {
-		return nil, fmt.Errorf("failed to decode error from page: %w", err)
-	}
-	return nil, fmt.Errorf("failed to query explorer network: %v", errResult)
-}
-
-func (a *App) query(queryString string, result interface{}) error {
-	response, err := a.baseQuery(queryString)
-	if err != nil {
-		return err
+	for param, prop := range strs {
+		value := r.URL.Query().Get(param)
+		if value != "" {
+			*prop = &value
+		}
 	}
-	defer response.Close()
-
-	if err := json.NewDecoder(response).Decode(result); err != nil {
-		return err
+	trueVal := true
+	for param, prop := range bools {
+		value := r.URL.Query().Get(param)
+		if value == "true" {
+			*prop = &trueVal
+		}
 	}
-
-	return nil
-}
-
-func getOffset(ctx context.Context) int {
-	return ctx.Value(offsetKey{}).(int)
-}
-
-func getMaxResult(ctx context.Context) int {
-	return ctx.Value(maxResultKey{}).(int)
-}
-
-func getSpecificFarm(ctx context.Context) string {
-	return ctx.Value(specificFarmKey{}).(string)
-}
-
-func getIsGateway(ctx context.Context) string {
-	return ctx.Value(isGatewayKey{}).(string)
-}
-
-func calculateMaxResult(r *http.Request) (int, error) {
-	maxResultPerpage := r.URL.Query().Get("size")
-	if maxResultPerpage == "" {
-		maxResultPerpage = "50"
+	farmIDs := strings.Split(r.URL.Query().Get("farm_ids"), ",")
+	if len(farmIDs) != 1 || farmIDs[0] != "" {
+		filter.FarmIDs = make([]uint64, len(farmIDs))
+		for idx, id := range farmIDs {
+			parsed, err := strconv.ParseUint(strings.TrimSpace(id), 10, 64)
+			if err != nil {
+				return filter, limit, errors.Wrap(ErrBadRequest, fmt.Sprintf("couldn't parse farm_id indexed at %d %s", idx, err.Error()))
+			}
+			filter.FarmIDs[idx] = parsed
+		}
 	}
-
-	maxResult, err := strconv.Atoi(maxResultPerpage)
-	if err != nil {
-		return 0, fmt.Errorf("invalid page number : %w", err)
+	if strings.Contains(fmt.Sprint(r.URL), "gateways") {
+		filter.Domain = &trueVal
 	}
-
-	return maxResult, nil
-}
-
-func calculateOffset(maxResult int, r *http.Request) (int, error) {
 	page := r.URL.Query().Get("page")
+	size := r.URL.Query().Get("size")
```

Comment:
I would move extracting the pagination information to a separate method so I can reuse it with other calls.

```
offset, limit := GetPage(r)
```
it checks the url params, and return the default value if no pagination informatin is there. Hence it can be called always from any list handler.